consensus/proto_array/src/proto_array_fork_choice.rs

```diff
@@ -193,10 +195,12 @@ impl ProtoArrayForkChoice {
             justified_checkpoint,
             finalized_checkpoint,
             execution_status,
+            unrealized_justified_checkpoint: Some(justified_checkpoint),
+            unrealized_finalized_checkpoint: Some(finalized_checkpoint),
```

Comment:
I agree with what you have said.

I also spent some time considering how this works with the optimisation to avoid recomputing unrealised FFG (UFFG):

```
parent_justified.epoch == block_epoch
  && parent_finalized.epoch + 1 >= block_epoch
```

I believe this optimisation will mean that we don't recompute UFFG for the rest of the epoch (we'll just use the anchor values). I think this is correct. Even if it's not I think it's acceptable since that for some epoch `N` to be finalized there must exist some chain of at least an epoch extending from it (two epochs, I think). This chain should be enough to get us out into the territory where we naturally computing UFFG again.

Reply:
Oh also if epoch `N` is finalized then there must be some epoch `> N` justified, so there's no need for forking in the first epoch of the chain after an anchor state.

I am convinced that what you have here is good!